### PR TITLE
Add support for HPP protocol

### PR DESCRIPTION
**Pull request by DaniElectra**
Implement proper handling for `HPPPacket` packets, as we need to omit the protocol ID on the `RMCResponse`.

This PR depends on https://github.com/PretendoNetwork/nex-go/pull/24

### CODE REVIEW

File: globals/respond_not_implemented.go [hunk 1]
```diff
@@ -3,29 +3,46 @@ package globals
...
+		responsePacket.SetSource(packet.Destination())
+		responsePacket.SetDestination(packet.Source())
+		responsePacket.SetType(nex.DataPacket)
+		responsePacket.SetPayload(rmcResponseBytes)
+
+		responsePacket.AddFlag(nex.FlagNeedsAck)
+		responsePacket.AddFlag(nex.FlagReliable)
+
+		client.Server().Send(responsePacket)
+	}
```
Comment (jonbarrow) [inline globals/respond_not_implemented.go:47]: Minor thing, but a good chunk of these branches is duplicate code. Such as getting the sender, getting the request, setting the error and payload, etc

To minimize duplicate code those pieces should probably happen outside this check, and only handle the real type-specific stuff in each branch

File: globals/respond_not_implemented_custom.go [hunk 2]
```diff
@@ -3,30 +3,48 @@ package globals
...
+		responsePacket.SetSource(packet.Destination())
+		responsePacket.SetDestination(packet.Source())
+		responsePacket.SetType(nex.DataPacket)
+		responsePacket.SetPayload(rmcResponseBytes)
+
+		responsePacket.AddFlag(nex.FlagNeedsAck)
+		responsePacket.AddFlag(nex.FlagReliable)
+
+		client.Server().Send(responsePacket)
+	}
```
Comment (jonbarrow) [inline globals/respond_not_implemented_custom.go:48]: Same comment as before about duplicate code

**Review by jonbarrow — CHANGES_REQUESTED** [review]

**Comment by DaniElectra** [comment]
I tried to reduce the duplicates as much as possible. The RMC error is still duplicate since I need to use the value from the switch case on the area that it encloses

**Comment by jonbarrow** [comment]
Looks good! I just merged your PR in `nex-go` as well as made a new release, so once that's updated here I'll merge

**Comment by DaniElectra** [comment]
> Looks good! I just merged your PR in `nex-go` as well as made a new release, so once that's updated here I'll merge

Done!

**Pull request merged**